DatadogInternal/Sources/NetworkInstrumentation/URLSession/URLSessionTaskDelegate+Tracking.swift

```diff
@@ -0,0 +1,28 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+
+private var firstPartyHostsKey: UInt8 = 41
+private var coreKey: UInt8 = 43
+
+public extension URLSessionDelegate {
+     /// Returns the `DatadogCore` for this delegate.
+    var core: DatadogCoreProtocol? {
+        set {
+            objc_setAssociatedObject(self, &coreKey, newValue, objc_AssociationPolicy.OBJC_ASSOCIATION_RETAIN_NONATOMIC)
```

Comment:
question/ Does it create strong reference to `core` in `URLSessionDelegate`? That would mean `core` is no longer managed (solely) by `CoreRegistry`, hence we won't be able to deallocate it with `stop()` / `deinitialize()` APIs.

Reply:
`OBJC_ASSOCIATION_ASSIGN` should be weak reference and must not increment the retain 
I updated it.